sdk/core/Azure.Core/src/Internal/ArrayBackedPropertyBag.cs

```diff
@@ -11,33 +13,56 @@ namespace Azure.Core
     /// If the item count is less than 2, there are no allocations. Any additional items are stored in an array which will grow as needed.
     /// MUST be passed by ref only.
     /// </summary>
-    internal struct ArrayBackedPropertyBag
+    internal struct ArrayBackedPropertyBag<TKey, TValue> where TKey : struct, IEquatable<TKey>
     {
-        private KVP _first;
-        private KVP _second;
-        private KVP[] _rest;
+        private Kvp _first;
+        private Kvp _second;
+        private Kvp[] _rest;
         private int _count;
 
-        private readonly struct KVP
+        private readonly struct Kvp
```

Comment:
Interesting, why not use KeyValuePair which is also a readonly struct.

Reply:
Results with KeyValuePair are about 15% worse across all tests